src/annotator/util/scroll.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Return a promise that resolves on the next animation frame.
+ */
+function nextAnimationFrame() {
+  return new Promise(resolve => {
+    requestAnimationFrame(resolve);
+  });
+}
+
+/**
+ * Linearly interpolate between two values.
+ *
+ * @param {number} a
+ * @param {number} b
+ * @param {number} fraction - Value in [0, 1]
+ */
+function interpolate(a, b, fraction) {
+  return a + fraction * (b - a);
+}
+
+/**
+ * Return the offset of `element` from the top of a positioned ancestor `parent`.
+ *
+ * @param {HTMLElement} element
+ * @param {HTMLElement} parent - Positioned ancestor of `element`
+ * @return {number}
+ */
+export function offsetRelativeTo(element, parent) {
+  let offset = 0;
+  while (element !== parent && parent.contains(element)) {
+    offset += element.offsetTop;
+    element = /** @type {HTMLElement} */ (element.offsetParent);
+  }
+  return offset;
+}
+
+/**
+ * Scroll `element` until its `scrollTop` offset reaches a target value.
+ *
+ * @param {Element} element - Container element to scroll
+ * @param {number} offset - Target value for the scroll offset
+ * @param {object} options
+ *   @param {number} [options.maxDuration]
+ * @return {Promise<void>} - A promise that resolves once the scroll animation
+ *   is complete
+ */
+export async function scrollElement(
+  element,
+  offset,
+  { maxDuration = 500 } = {}
+) {
+  const initialOffset = element.scrollTop;
+  const targetOffset = offset;
+  const scrollStart = Date.now();
+
+  // Choose a scroll duration proportional to the scroll distance, but capped
+  // to avoid it being too slow.
+  const pixelsPerMs = 3;
```

Comment:
I think it makes sense since `Ms` is an abbreviation rather than an acryonym. An alternative would be to use "millis" instead, but we do use `ms` quite widely as a shorthand.